src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -416,10 +416,20 @@ static void PaintStep(Visual visual, in PaintingSession session)
 
 		static void PostPaintingClipStep(Visual visual, SKCanvas canvas)
 		{
+#if DEBUG
+			canvas.Save();
 			if (visual.GetPostPaintingClipping() is { } postClip)
 			{
 				canvas.ClipPath(postClip, antialias: true);
 			}
+
+			var nonOptimizedClip = (canvas.DeviceClipBounds, canvas.IsClipRect);
+			canvas.Restore();
+#endif
+			visual.ApplyPostPaintingClipping(canvas);
+#if DEBUG
+			Debug.Assert(nonOptimizedClip.IsClipRect == canvas.IsClipRect && nonOptimizedClip.DeviceClipBounds == canvas.DeviceClipBounds);
+#endif
```

Comment:
All methods on `Debug` class are conditional to `DEBUG`, no needs to add it again

```suggestion
			Debug.Assert(nonOptimizedClip.IsClipRect == canvas.IsClipRect && nonOptimizedClip.DeviceClipBounds == canvas.DeviceClipBounds);
```